### PR TITLE
Common Kolmogorov-Smirnov test

### PR DESCRIPTION
**Pull request by AlexanderSaydakov**
Should work for both classic quantiles and KLL

### CODE REVIEW

File: common/include/kolmogorov_smirnov_impl.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ namespace datasketches {
 
 template<typename Sketch>
 double kolmogorov_smirnov::delta(const Sketch& sketch1, const Sketch& sketch2) {
-  using Comparator = typename Sketch::comparator;
+  auto comparator = sketch1.get_comparator(); // assuming the same comparator in sketch2
```
Comment (jmalkin) [inline common/include/kolmogorov_smirnov_impl.hpp:27]: should we care if sketch1 and sketch2 have different comparators?

File: quantiles/include/quantiles_sketch_impl.hpp [hunk 2]
```diff
@@ -262,7 +262,7 @@ void quantiles_sketch<T, C, A>::serialize(std::ostream& os, const SerDe& serde)
 template<typename T, typename C, typename A>
 template<typename SerDe>
 auto quantiles_sketch<T, C, A>::serialize(unsigned header_size_bytes, const SerDe& serde) const -> vector_bytes {
-  const size_t size = get_serialized_size_bytes() + header_size_bytes;
+  const size_t size = get_serialized_size_bytes(serde) + header_size_bytes;
```
Comment (jmalkin) [inline quantiles/include/quantiles_sketch_impl.hpp:265]: good catch.. thx

File: quantiles/include/quantiles_sketch.hpp [hunk 3]
```diff
@@ -146,14 +146,13 @@ namespace quantiles_constants {
...
-          typename Comparator = std::less<T>,
+          typename Comparator = std::less<T>, // strict weak ordering function (see C++ named requirements: Compare)
           typename Allocator = std::allocator<T>>
 class quantiles_sketch {
 public:
-  using C = Comparator;
-  using A = Allocator;
-  using AllocDouble = typename std::allocator_traits<Allocator>::template rebind_alloc<double>;
-  using vector_double = std::vector<double, AllocDouble>;
+  using value_type = T;
```
Comment (jmalkin) [inline quantiles/include/quantiles_sketch.hpp:153]: i don't think this is actually used, but i guess it gives us a consistent API and makes it easy to access

**Review by jmalkin — APPROVED** [review]

File: common/include/kolmogorov_smirnov_impl.hpp (hunk 1 above)
Comment (AlexanderSaydakov) [reply common/include/kolmogorov_smirnov_impl.hpp:27]: right now they are passed as a type, not instance. sketch1 and sketch2 have the same type. potentially we could have two template parameters to support sketches of different types. I am not sure we need this

File: quantiles/include/quantiles_sketch.hpp (hunk 3 above)
Comment (AlexanderSaydakov) [reply quantiles/include/quantiles_sketch.hpp:153]: I tied to follow some practices of STL. they have many type aliases like this for users

**Pull request merged**